### ISSUE TITLE
Renaming an auto incrementing primary key makes the resulting column a non auto incrementing column

### ISSUE THREAD

**Issue by jimmymbiyu** [knex/knex#2767]
# Environment

Knex version: 0.14.6
Database + version: MySQL v14.14
OS: Ubuntu 14.04.5 LTS

# Bug
@smorey2 

I would like to rename an existing table's auto-incrementing primary key. I have the following code in the knex migration.
```
exports.up = function(knex, Promise) {
    return knex.schema.table('mobile_payments', table => table.renameColumn('transaction_id', 'id'));
};
```
The migration runs successfully without any errors and the PK is renamed as expected. However, it is no longer an auto-incrementing column.

I tried to chain an `alterTable` promise after the `renameColumn` promise as shown below but that did not work.
```
exports.up = function(knex, Promise) {
    return knex.schema.table('mobile_payments', table => table.renameColumn('transaction_id', 'id'))
        .then(() => knex.schema.alterTable('mobile_payments', table => table.increments('id').alter()));
};
```
This resulted in the error below.
```
Error: ER_MULTIPLE_PRI_KEY: Multiple primary key defined
```

**Comment by kibertoad** [comment]
Similar issues: #2703, #1783.

How is auto-incrementation implemented in MySQL? Is there a trigger created for that?

**Comment by radiumrasheed** [comment]
any workaround for this?

**Comment by kibertoad** [comment]
@radiumrasheed Well, if you contribute failing test for this, I could fix it :)

**Comment by radiumrasheed** [comment]
sure, a guide will help

**Comment by kibertoad** [comment]
@radiumrasheed Take a look at https://github.com/tgriesser/knex/blob/master/test/integration/migrate/index.js

To execute tests only on a single DB, run `cross-env DB=mysql npm run plaintest`

**Comment by gpetrov** [comment]
is there eventually a solution to this @kibertoad ? 

Just to do simple rename on a column created initially as incremental?

**Comment by kibertoad** [comment]
@gpetrov Which DB are you using?

**Comment by gpetrov** [comment]
Well MySQL but it seems it is a common problem.
On column rename or type change the incremental is lost and there is no way to add it back.

**Comment by kibertoad** [comment]
Released in 0.95.0, please reopen if you still experience the issue.
